### PR TITLE
test: some cleanup re: #12890

### PR DESCRIPTION
**Pull request by vkarpov15**
**Summary**

Applied some suggestions from #12890. Mostly getting rid of unnecessary `connectionsToClose` - left it in one case where it makes sense. Also refactored out some `Model.watch()` cases, and fixed a potential bug due to the MongoDB node driver continuing to poll after change stream is closed.

**Examples**

### CODE REVIEW

File: test/collection.capped.test.js [hunk 1]
```diff
@@ -55,7 +54,6 @@ describe('collections: capped:', function() {
 
   it('skips when setting autoCreate to false (gh-8566)', async function() {
     const db = start();
```
Comment (hasezoey) [inline test/collection.capped.test.js:56]:
```suggestion
    db = start();
```

File: test/collection.test.js [hunk 2]
```diff
@@ -8,15 +8,18 @@ const assert = require('assert');
 const mongoose = start.mongoose;
 
 describe('collections:', function() {
-  const connectionsToClose = [];
+  let db = null;
 
-  after(async function() {
-    await Promise.all(connectionsToClose.map((v) => v.close()));
+  this.afterEach(async function() {
```
Comment (hasezoey) [inline test/collection.test.js:13]:
```suggestion
  afterEach(async function() {
```

i dont think `this.` is required here, removing it to make it consistent with other test files

File: test/model.test.js [hunk 3]
```diff
@@ -4336,7 +4326,6 @@ describe('Model', function() {
   it('save max bson size error with buffering (gh-3906)', async function() {
     this.timeout(10000);
     const db = start({ noErrorListener: true });
```
Comment (hasezoey) [inline test/model.test.js:4328]: from what i can tell, in this file / tests the connections do not get properly closed if the test is not successful (which `connectionsToClose` would solve)

**Review by hasezoey — CHANGES_REQUESTED** [review]
mostly looks good to me, though there are some places that have not been changed yet, or were changed back to not properly close in case of test failure

**Review by IslandRhythms — APPROVED** [review]
LGTM

**Pull request merged**